src/Adapter/Carrier/CommandHandler/EditCarrierHandler.php

```diff
@@ -75,20 +75,35 @@ public function handle(EditCarrierCommand $command): CarrierId
         if ($command->getLocalizedDelay()) {
             $carrier->delay = $command->getLocalizedDelay();
         }
+        if ($command->getMaxWidth()) {
+            $carrier->width = $command->getMaxWidth();
+        }
+        if ($command->getMaxHeight()) {
+            $carrier->height = $command->getMaxHeight();
+        }
+        if ($command->getMaxDepth()) {
+            $carrier->depth = $command->getMaxDepth();
+        }
+        if ($command->getMaxWeight()) {
+            $carrier->weight = $command->getMaxWeight();
+        }
 
         $this->carrierValidator->validate($carrier);
 
-        $carrierId = $this->carrierRepository->updateInNewVersion($command->getCarrierId(), $carrier);
-
+        $newCarrier = $this->carrierRepository->updateInNewVersion($command->getCarrierId(), $carrier);
+        if ($command->getGroupAccess()) {
+            $this->carrierValidator->validateGroupsExist($command->getGroupAccess());
```

Comment:
The validation should be done earlier, because in case this is invalid but other properties are not the handler will have a dual behaviour where one part of the data is saved and the other isn't which would be hard to understand for users